libtransmission/rpcimpl.cc

```diff
@@ -853,7 +853,7 @@ void initField(tr_torrent const* const tor, tr_stat const* const st, tr_variant*
             tr_variantInitList(initme, n);
             for (tr_file_index_t i = 0; i < n; ++i)
             {
-                tr_variantListAddBool(initme, tr_torrentFile(tor, i).wanted);
+                tr_variantListAddInt(initme, tr_torrentFile(tor, i).wanted);
```

Comment:
For `initField()` -- The horses are already out of the gate at this point. Since 4.0.0 and 4.0.1 are released, there are stable versions of Transmission out there that use bools and that use ints for these fields. At this point, changing it _back_ so that we say "it's always an int, except for 4.0.0 and 4.0.1 where it's a bool" is going to be confusing to explain to people who come looking for support answers later.

TBH I think the simpler, easier-to-understand fix at this point might be to treat this as a bug in the section 5 of the spec that should say "this was always documented as an array of booleans; but prior to 4.0.0, it was serialized as an array of `0` or `1`.

Reply:
just imo, but it doesn't seem like Tr4 has so much distribution that this should not revert.  Unfortunately some of the client-side tools that have this legacy dependency are not trivially fixable (ie they are distributed as Windows binaries, APK's, etc).

my mentality on these sorts of things is very Linus-like.  Usermode compat is job#1, regardless of the quirks that have to be carried forward. In this case, RPC clients can be thought of as "Usermode". The exceptions to this usually involve security and I can't see that in this case.

This raises a whole other issue around RPC versioning and some form of contracts, but that is much to big for this niggle.

Reply:
Yeah ok I'll buy that, at least for this issue.

Please update rpc-spec.md to say this has always been implemented as an int but before 4.0.0 was documented as a bool. Probably should be listed in section 5.

Reply:
Just the doc note for 4.0.0